ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx

```diff
@@ -61,11 +76,39 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
   );
 
   const onSaveButtonClick = () => {
+    const { newVariables, isSelectedVariablesUpdated } = updateVariableDefaultValues(savedVariables, variableValues);
+
+    const isTimeRangeUpdated = isRelativeTimeRange(timeRange) && dashboard.spec.duration !== timeRange.pastDuration;
+
+    // Save dashboard if active timeRange from plugin-system is relative and different than currently saved
+    if (isTimeRangeUpdated || isSelectedVariablesUpdated) {
+      openSaveChangesConfirmationDialog({
+        onSaveChanges: (saveDefaultTimeRange, saveDefaultVariables) => {
+          if (isRelativeTimeRange(timeRange) && saveDefaultTimeRange === true) {
+            dashboard.spec.duration = timeRange.pastDuration;
+          }
+          if (saveDefaultVariables === true) {
+            dashboard.spec.variables = newVariables;
+          }
+          saveDashboard();
+          refresh();
```

Comment:
is there a reason why we need to call `refresh` here?

Reply:
Removed this, not needed anymore after updates in TemplateVariableProvider.tsx